packages/build-standalone/src/workflow.ts

```diff
@@ -0,0 +1,50 @@
+export type WorkflowStage = 'validate' | 'cleanup';
+
+export type WorkflowHandler<T> = (context: T) => Promise<T> | T;
+
+export class Workflow<T> {
+  #events = new Map<WorkflowStage, Array<WorkflowHandler<T>>>();
+
+  #handlers(phase: WorkflowStage): WorkflowHandler<T>[] {
+    return this.#events.get(phase) || [];
+  }
+
+  register<P extends WorkflowStage>(
+    phase: P,
+    handler: WorkflowHandler<T>,
+  ): this {
+    this.#events.set(phase, [...this.#handlers(phase), handler]);
+    return this;
+  }
+
+  async execute(context: T): Promise<T> {
+    return await series(context, [
+      (c) => series(c, this.#handlers('validate')),
```

Comment:
Not clear why `validate` - it seems we're actually doing all the work in the `validate` step in the example plugins.

Reply:
This is the definition of the order of the phases referenced in the plugins... it could be rewritten to:

```typescript
const phases: WorkflowStage[] = ['validate', 'cleanup'];

return await series(context, 
  phases.map(phase => (c) => series(c, this.#handlers(phase))
);
```